server/index.js

```diff
@@ -4,116 +4,147 @@ const app = express();
+const { EDESTADDRREQ } = require('constants');
 
 app.use(express.static('public'));
 app.use(bodyParser.urlencoded({ extended: true }));
 app.use(bodyParser.json());
 app.use(cors());
 
+//TODO: Refactor to have routes, models, and middelware to handle errors
+
 //returns an object that includes price, promotion/discount
 app.get('/PriceAndPromotion/:product_id', (req, res) => {
-
-  let id = req.params.product_id;
-
-  PriceAndPromo.find({ product_id: id })
-    .then(game => 
-      {
-      if (!game) {
-        res.status(400).send('No game to return');
-      } else {
-        let data = {
-          price: game[0].price,
-          promotion: game[0].discount,
+ 
+  let id = (req.params.product_id);
+  if(isNaN(id)){
+    res.status(400).send('Invalid Product Id');
+  } else {
+    id = [Number(req.params.product_id)];
+   
+    getPriceAndPromotion(id)
+      .then( (response) => {
+       
+        let {base_price, discount, max} = response[0];
+        let price = base_price;
+        let promotion = base_price * (discount/100);
+        promotion = Number(promotion.toFixed(2));
+
+        //check for general or publisher discount to calculate new price for UI;
+        if(max) {
+          price *= (1 - max/100)
+          price = Number(price.toFixed(2));
         }
-        res.status(200).send(data);
-      }
-    })
-    .catch(err => {
-      res.status(404).send(err);
-    });
+
+        res.status(200).send({price, promotion});
+      })
+      .catch( (err) => {
+        res.status(404).send('Data Not Found');
+        console.log(err);
+      })
+  }
 });
 
 //returns price and promos for an array of product ids
-app.get('/PriceAndPromotion/multiple/:product_ids', (req, res) => {
-  let arrayOfIds = JSON.parse(req.params.product_ids);
-  let dataBundle = async () => {
-    let priceAndPromosArray = []
-    for (let i = 0; i < arrayOfIds.length; i++) {
-      await PriceAndPromo.find({ product_id: arrayOfIds[i] })
-        .then((doc) => {
-          let data = {
-            product_id: doc[0].product_id,
-            price: doc[0].price,
-            promotion: doc[0].discount
-          }
-          priceAndPromosArray.push(data);
-        })
-        .catch(err => console.log(err))
+app.get('/PriceAndPromotion/multiple/:product_ids', async (req, res) => {
+
+  try {
+    if(!req.params.product_ids){
+      res.status(400).send('No Product Id');
+    } else if (req.params.product_ids) {
+      
+      let arrayOfIds = JSON.parse(req.params.product_ids);
+          
+      let arrayOfRecords = await getPriceAndPromotion(arrayOfIds);
+
+      let arrayOfPriceandPromtions = await arrayOfRecords.map( record => {
+
+        let {id, base_price, discount, max} = record;
+        let product_id = id;
+        let price = base_price;
+        let promotion = base_price * (discount/100);
+        promotion = Number(promotion.toFixed(2));
+        
+        if(max) {
```

Comment:
I noticed some inconsistent spacing that a linter would fix (lines 61, 63, here)